### PR TITLE
feat(goto): assume http:// for localhost navigations

### LINKED ISSUE #803: [BUG] browserType chromium and webkit produces error on schemeless url
**Context:**
- PlayWright Version: 0.10.0
- Operating System: Mac OS X - 10.14.6
- Extra: Node v13.7.0

**Code Snippet**
```javascript
const playwright = require("playwright");

(async () => {
    for (const browserType of ["chromium", "firefox", "webkit"]) {
        const browser = await playwright[browserType].launch()
        const context = await browser.newContext({})
        try {
            const page = await context.newPage("localhost:8080")
            console.log("Ok", browserType)
        } catch {
            console.log("Err", browserType, e)
        }

        await browser.close()
    }
})();
```

**Describe the bug**

For browserType **"chromium"** and **"webkit"** url passed to newPage fails if scheme not provided. Expect playwright to assume scheme to be http for browserType "chromium" and "webkit".

<details>
<summary>stdout</summary>
<pre>
Err chromium Error: Navigation to localhost:8080 failed: net::ERR_ABORTED; maybe frame was detached?
    at LifecycleWatcher._onAbortedNewDocumentNavigation (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/frames.js:879:49)
    at FrameManager.requestFailed (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/frames.js:176:29)
    at CRNetworkManager._onLoadingFailed (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/chromium/crNetworkManager.js:235:34)
    at CRSession.emit (events.js:321:20)
    at /Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/chromium/crConnection.js:135:47
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
  -- ASYNC --
    at Frame.<anonymous> (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/helper.js:54:23)
    at Page.goto (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/page.js:215:33)
    at Page.<anonymous> (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/helper.js:55:31)
    at BrowserContext.newPage (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/browserContext.js:48:24)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /Users/aes/sandbox/playwright-test/src/main.spec.js:26:26
  -- ASYNC --
    at BrowserContext.<anonymous> (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/helper.js:54:23)
    at /Users/aes/sandbox/playwright-test/src/main.spec.js:26:40
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
Ok firefox
Err webkit Error: Protocol error (Browser.navigate): Navigation cancelled [object Object]
    at /Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/webkit/wkConnection.js:99:63
    at new Promise (<anonymous>)
    at WKSession.send (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/webkit/wkConnection.js:98:16)
    at WKPage.navigateFrame (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/webkit/wkPage.js:262:79)
    at navigate (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/frames.js:343:61)
    at Frame.goto (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/frames.js:350:13)
    at Frame.<anonymous> (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/helper.js:55:31)
    at Page.goto (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/page.js:215:33)
    at Page.<anonymous> (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/helper.js:55:31)
    at BrowserContext.newPage (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/browserContext.js:48:24)
  -- ASYNC --
    at Frame.<anonymous> (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/helper.js:54:23)
    at Page.goto (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/page.js:215:33)
    at Page.<anonymous> (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/helper.js:55:31)
    at BrowserContext.newPage (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/browserContext.js:48:24)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /Users/aes/sandbox/playwright-test/src/main.spec.js:26:26
  -- ASYNC --
    at BrowserContext.<anonymous> (/Users/aes/sandbox/playwright-test/node_modules/playwright-core/lib/helper.js:54:23)
    at /Users/aes/sandbox/playwright-test/src/main.spec.js:26:40
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  message: 'Protocol error (Browser.navigate): Navigation cancelled [object Object]'
}
</pre>
</details>

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes #803

### CODE REVIEW

File: test/navigation.spec.js [hunk 1]
```diff
@@ -28,6 +28,10 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
       await page.goto(server.EMPTY_PAGE);
       expect(page.url()).toBe(server.EMPTY_PAGE);
     });
+    fit('should use http for no protocol', async({page, server}) => {
```
Comment (dgozman) [inline test/navigation.spec.js]: Shouldn't this test localhost url? Also, no fit please :)

File: src/helper.ts [hunk 2]
```diff
@@ -211,6 +211,12 @@ class Helper {
     tokens.push('$');
     return new RegExp(tokens.join(''));
   }
+
+  static completeUserURL(urlString: string): string {
+    if (urlString.startsWith('localhost') || urlString.startsWith('127.0.0.1'))
```
Comment (dgozman) [inline src/helper.ts]: Perhaps allow `0:0:0:0:0:0:0:1` and `::1` as well?

**Review by dgozman — APPROVED** [review]

File: test/navigation.spec.js (hunk 1 above)
Comment (aslushnikov) [reply test/navigation.spec.js]: Yeah, bot's aren't happy with fits!

File: src/helper.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/helper.ts]: I'd strongly encourage using URLs where urls are required. localhost and 127.0.0.1 are too wide-spread to ignore, but ipv6 is not (yet).

**Pull request merged**